### PR TITLE
Support for LoadBalancerIP annotation

### PR DESCRIPTION
**Pull request by saurabhvaidvmware**

### CODE REVIEW

File: internal/nodes/avi_model_advl4_translator.go [hunk 1]
```diff
@@ -447,7 +447,11 @@ func (o *AviObjectGraph) ConstructSharedVipSvcLBNode(sharedVipKey, namespace, ke
 		}
 
 		if i == 0 {
-			sharedPreferredVIP = svcObj.Spec.LoadBalancerIP
+			if svcObj.Spec.LoadBalancerIP != "" {
+				sharedPreferredVIP = svcObj.Spec.LoadBalancerIP
+			} else if svcObj.Spec.LoadBalancerIP == "" && svcObj.Annotations[lib.LoadBalancerIP] != "" {
```
Comment (akshayhavile) [inline internal/nodes/avi_model_advl4_translator.go:452]: @saurabhvaidvmware: I think we can avoid first check `svcObj.Spec.LoadBalancerIP== ""`. This should be done everywhere.

File: internal/nodes/dequeue_ingestion.go [hunk 2]
```diff
@@ -508,16 +508,28 @@ func handleL4SharedVipService(namespacedVipKey, key string, fullsync bool) {
 		// Initializing the preferred VIP from the first Service we get, so any other Service
 		// that wishes for static IP allocation differently conflicts with this.
 		if i == 0 {
-			sharedVipLBIP = svcObj.Spec.LoadBalancerIP
+			if svcObj.Spec.LoadBalancerIP != "" {
```
Comment (swathinsankaran) [inline internal/nodes/dequeue_ingestion.go:511]: these conditions are repeated everywhere. It will be better if we can move this to a function.

Comment (saurabhvaidvmware) [reply internal/nodes/dequeue_ingestion.go]: Added functions in util.go to handle condition checks

File: internal/nodes/avi_model_advl4_translator.go (hunk 1 above)
Comment (saurabhvaidvmware) [reply internal/nodes/avi_model_advl4_translator.go]: Dropped this condition check.

**Comment by DixitAakash** [comment]
build ako

**Review by akshayhavile — APPROVED** [review]

**Pull request merged**